tests/test_interfaces.py

```diff
@@ -778,6 +723,103 @@ def test_get_interface_mac_address(self):
 
         self.assertEqual(self.network_manager.get_interface_mac(interface_name), self.mac_address)
 
+    def test_get_interface_1_ap_1_mon_internet_interfaces(self):
```

Comment:
Now I see what you were thinking. Unit testing ignores the big picture and for example here you would write a test case for each possible case. In this example you would need one to check the AP mode, 1 for monitor mode, and one for internet.